docs/blog/2023-09-06-distributed-key-generation.md

```diff
@@ -0,0 +1,154 @@
+
+The most famous secret-sharing scheme was introduced by Adi Shamir in 1979 and is known as Shamir’s Secret Sharing Scheme. The idea behind this involves polynomial interpolation and more specifically [Lagrange Interpolation](https://en.wikipedia.org/wiki/Lagrange_polynomial). (Mathematics 😞)  
+
+In a nutshell: 
+
+1. For a given secret s, a polynomial φ of degree t-1, where t is the threshold, is chosen so that the constant term of this polynomial, the value φ(0) in other words, is the secret.
+2. Then each participant gets a share, which is a point on this polynomial.
+3. In order to reconstruct the polynomial, and reveal the secret we need t participants to be involved. Otherwise, the secret cannot be revealed.
+
+The above works under the context that we have a certain entity, named the Dealer that is responsible for the sharing. However thinking that the dealer is an honest entity, is not the best assumption. The dealer still knows the secret and the polynomial, and thus becomes a single point of failure. 
+
+# DKGs:
+
+As previously outlined, secret-sharing methods rely on a single dealer who is privy to the secret, posing a potential security risk. This vulnerability can however be addressed by Distributed Key Generation (DKG), which allows each participant to contribute equally (⚖️) to the secret-sharing process. 
+
+To demonstrate this, let’s go back to our pirate example. 
+
+Let’s say that we have Pirate A, B C, and we cannot forget Jack. We can say now that the secret is the actual gold of the treasure box, and the box is protected with a secret code, the key. The distribution now involves all four of them to contribute to the randomness of the key. 
+
+**🏴‍☠️ First Step:** Each of the pirates (Pirates A, B, C, and Jack) agrees to jointly create a secret key to a treasure box. But none of them should have the complete key on their own.
+
+**🏴‍☠️ Generation:** Jack creates a polynomial of a degree t and distributes shares to the other pirates including himself. A does the same, and B and C follow too.
+
+That means that at the end, each pirate will have four shares created including his/her own.
+
+🙍🏾Jack : {s_Jack, s_A, s_B, s_C}    🙍🏾‍♂️A: { s_A, s_Jack, s_B, s_C}
+
+🙍🏾‍♀️B: { s_B, s_Jack, s_A, s_C}         🥷🏾C: {s_C, s_Jack, s_A, s_B}
+
+**🏴‍☠️ Sharing:** As soon as the above is complete, each pirate distributes their shares to the other pirates. So, at the end of this, each should have a share from the others. 
```

Comment:
"So, at the end of this, each should have a share from the others" - this could be confusing to the uninitiated; worth clarifying that every pirate will have one share from each of the other pirates (plus their own)